bot.py

```diff
@@ -296,6 +304,29 @@ async def create_or_get_voice_client(self, message):
 
         return self.voice_client
 
+    @tasks.loop(seconds=2)
+    async def disconnect_checker(self):
+        """
+        Checks every few seconds whether the voice_client is playing something in a
+        voice channel. If it's not then the bot will be automatically disconnected.
+        """
+        if self.voice_client and not self.voice_client.is_playing():
+            async with self.command_lock:
+                time_since_last_command = time.time() - self.last_command_time
+
+                if (
+                    self.voice_client.is_paused()
+                    and time_since_last_command < MusicBot.DISCONNECT_PAUSE_TIMER
+                ):
+                    return
+
+                if time_since_last_command < MusicBot.DISCONNECT_STOP_TIMER:
+                    return
+
+                self._stop()
+                await self.voice_client.disconnect()
+                self.voice_client = None
+
```

Comment:
Missing check for end-of-queue disconnect. I think the bot  `is_playing() == False` at the end of queue, so it will never disconnect upon reaching end-of-queue.

Reply:
is it possible that you misread
```python
if self.voice_client and not self.voice_client.is_playing():
```
as
```python
if self.voice_client and self.voice_client.is_playing():
```
?

Reply:
Yes, sorry.